### PR TITLE
avcpp: add version 2.1.0 and suppport conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **avcpp/***

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/avcpp/all/conandata.yml [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
+  "2.1.0":
+    url: "https://github.com/h4tr3d/avcpp/archive/refs/tags/v2.1.0.tar.gz"
+    sha256: "8398217dccb9f5b4cbb41e5bf4f73f47b461ed3ba8c3aefdda9f9dd714649855"
   "cci.20220301":
     url: "https://github.com/h4tr3d/avcpp/archive/fd4bc4662eb39853de8fcac4a663bebd0eea30b8.tar.gz"
     sha256: "e48eae2ec154bc69aed16159c8b18c9ffb4925ba672b022e94a3c9b96782a4bf"
 
 patches:
+  "2.1.0":
+    - patch_file: "patches/2.1.0-fix-cmake.patch"
```
Comment (prince-chrismc) [inline recipes/avcpp/all/conandata.yml:11]: patch data fields please 🙏

File: recipes/avcpp/all/conanfile.py [hunk 2]
```diff
@@ -51,56 +50,66 @@ def configure(self):
...
-        self.cpp_info.filenames["cmake_find_package"] = "avcpp"
-        self.cpp_info.filenames["cmake_find_package_multi"] = "avcpp"
-        self.cpp_info.names["cmake_find_package"] = "avcpp"
-        self.cpp_info.names["cmake_find_package_multi"] = "avcpp"
+        self.cpp_info.set_property("cmake_target_name", f"avcpp::{target_name}")
 
         self.cpp_info.components["AvCpp"].names["cmake_find_package"] = target_name
         self.cpp_info.components["AvCpp"].names["cmake_find_package_multi"] = target_name
-        self.cpp_info.components["AvCpp"].set_property("cmake_target_name", "avcpp::{}".format(target_name))
+        self.cpp_info.components["AvCpp"].set_property(f"cmake_target_name", f"avcpp::{target_name}")
```
Comment (prince-chrismc) [inline recipes/avcpp/all/conanfile.py:103]: linter

```suggestion
        self.cpp_info.components["AvCpp"].set_property("cmake_target_name", f"avcpp::{target_name}")
```

**Review by prince-chrismc — COMMENTED** [review]
two tiny nits

File: recipes/avcpp/all/conandata.yml (hunk 1 above)
Comment (toge) [reply recipes/avcpp/all/conandata.yml]: Thank you for your review!
Fixed.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

**Pull request merged**